src/test/java/de/tum/in/www1/artemis/exam/StudentExamIntegrationTest.java

```diff
@@ -173,12 +173,12 @@ class StudentExamIntegrationTest extends AbstractSpringIntegrationBambooBitbucke
 
     private final List<LocalRepository> studentRepos = new ArrayList<>();
 
-    private static final int NUMBER_OF_STUDENTS = 5;
+    private static final int NUMBER_OF_STUDENTS = 2;
 
     @BeforeEach
     void initTestCase() throws Exception {
-        // the service already creates 5 students, 1 tutor, 1 editor, 1 instructor
-        programmingExerciseTestService.setupTestUsers(TEST_PREFIX, 0, 0, 0, 1);
+        userUtilService.addUsers(TEST_PREFIX, 3, 1, 0, 2);
```

Comment:
Did you consider also reducing the number of instructors? Only the test `testFindAllTestRunsForExam()` needs the instructor2. You could consider adding the second instructor only in the beginning of this test

Reply:
Adding users only for individual tests is slower than creating the user in the initTestCase method. Seems a bit weird, but I tested it and it works like that on my machine xD
The number of students was updated 🚀

Reply:
Oh I see, in this case it makes sense to create two in the before each method. Interesting behavior :sweat_smile: